graphql/provider.go

```diff
@@ -40,10 +62,66 @@ func graphqlConfigure(ctx context.Context, d *schema.ResourceData) (interface{},
 		GQLServerUrl:   d.Get("url").(string),
 		RequestHeaders: d.Get("headers").(map[string]interface{}),
 	}
+
+	if oauth2LoginQueryValueAttribute := d.Get("oauth2_login_query_value_attribute"); d.Get("oauth2_login_query") != "" && oauth2LoginQueryValueAttribute != "" {
```

Comment:
Maybe relax this check and use a default value for `oauth2_login_query_value_attribute`.

Reply:
Turns out I couldn't think of a default value that would make sense considering the lack of standardization of this way of performing OAuth 2.0 authentication with GraphQL queries/mutations.